R/ROADEF-R/tuning/parameters.txt
## Template for parameter description file for Iterated Race.
##
## The format is one parameter per line. Each line contains:
##
## 1: Name of the parameter. An unquoted alphanumeric string,
##    example: ants

## 2: Switch to pass the parameter. A quoted (possibly empty) string, 
##    if the value and the switch must be separated, add a space at
##    the end of the string. Example : "--version1 --ants "

## 3: Type. An unquoted single letter, among 
##     i: Integer, c: categorical, o: ordinal, r: real.

## 4: For c and o: All possible values, that is, a variable number of
##    quoted or unquoted strings separated by commas within
##    parenthesis. Empty strings and strings containing commas or
##    spaces must be quoted.
##    For i,r: a pair of unquoted numbers representing minimum and
##    maximum values. 

## 5: A conditional parameter can be defined according to the values of
##    one or several other parameters.  This is done by adding a
##    character '|' followed by an R expression involving the names of
##    other parameters. This expression must return TRUE if the 
##    condition is satisfied, FALSE otherwise.

# 1:            2:                   3: 4:      5:
g__timeLimit  "--g__timeLimit="   c   (600)
h__main_iter     "--h__main_iter="   i   (1, 100)
h__max_iter     "--h__max_iter="   i   (1, 1000)
h__max_candidates     "--h__max_candidates="   i   (1, 50)
h__temperature          "--h__temperature="          i,log  (1, 100000)
h__cooling_rate          "--h__cooling_rate="          r,log  (0.00005, 1.0)
h__cuts_prob          "--h__cuts_prob="          r  (0.0, 1.0)
h__clean_cuts_prob          "--h__clean_cuts_prob="          r  (0.0, 1.0)
h__try_rotation     "--h__try_rotation="   i   (0, 1)
rem__iterations_initial          "--rem__iterations_initial="          i  (10, 1000)
rem__iterations_remake          "--rem__iterations_remake="          i  (10, 100)
rem__max_no_improve          "--rem__max_no_improve="          i  (1, 100)
rem__prob_accept_worse_def          "--rem__prob_accept_worse_def="          r  (0, 0.5)
rem__prob_accept_worse          "--rem__prob_accept_worse="          r  (0, 0.5)
#h__weights__space        "--h__weights__space="        i,log  (1, 100000)
#h__weights__seq        "--h__weights__seq="        i,log  (1, 100000)
#h__weights__defects        "--h__weights__defects="        i,log  (1, 100000)
#rem__rotation
#rem__num_trees
#rem__probability
#h__rotation_probs            "--"                 c  ("x1" ,"x2", "x3")
#h__level_probs            "--paramreal="       r  (1.5, 4.5)



#h__temperature          "--param1 "          i  (1, 10) | mode %in% c("x1", "x2")
#h__cooling_rate          "--param2 "          i  (1, 10) | mode %in% c("x1", "x3") && real > 2.5 && real <= 3.5
#h__rotation_probs            "--"                 c  ("x1" ,"x2", "x3")
#h__level_probs            "--paramreal="       r  (1.5, 4.5)
#h__weights__space        "--mutation="        o  ("none", "very low", "low", "medium", "high", "very high", "all")
#unused         "-u "                c  (1, 2, 10, 20)


R/ROADEF-R/tuning/configurations.txt
## Template for specifying initial parameter configurations, for
## example, the default configuration, in irace.
##
## This filename must be specified via the --candidates-file command-line option
## (or candidatesFile in scenario.txt).
##
## The format is one parameter configuration per line, and one parameter per
## column. The first line gives the parameter name corresponding to
## each column (names must match those given in the parameters
## file). Each configuration must satisfy the
## parameter conditions (NA should be used for those parameters that
## are not enabled for a given configuration) and, if given, the
## constraints that describe forbidden configurations.
h__main_iter    h__max_iter h__max_candidates   h__temperature  h__cooling_rate h__cuts_prob    h__clean_cuts_prob  h__try_rotation rem__iterations_initial rem__iterations_remake  rem__max_no_improve rem__prob_accept_worse_def  rem__prob_accept_worse  
5       500     5   100     0.00005     0.5 0.8 1   100     10  10  0.02    0.05


python/package/params.py
def calculate_paths_root(root, data_set='A'):
    path_checker = root + "resources/checker/"
    return {
    'root': root
    ,'img': root + "img/"
    ,'latex': root + "latex/"
    ,'data': root + "resources/data/dataset_{}/".format(data_set)
    ,'checker': path_checker
    ,'checker_data': path_checker + "instances_checker/"
}


def calculate_paths_results(results):
    return {
    'results': results
    ,'experiments': results + "test/"
    }

base = '/home'
# base = 'C:/Users'
root = base + '/pchtsp/Documents/projects/ROADEF2018/'
results = base + '/pchtsp/Dropbox/ROADEF2018/'

PATHS = {**calculate_paths_root(root), **calculate_paths_results(results)}

OPTIONS = {
    'timeLimit': 600
    , 'gap': 0.1
    , 'solver': "HEUR"  # CPLEXPY, CPLEX_CMD, GUROBI, HEUR
    , 'integer': False
    , 'input_path': PATHS['experiments']
    , 'output_path': PATHS['experiments']
    , 'output_file_name': None
    , 'case_name': 'A15'
    , 'max_plates': 10
    , 'max_width': 6000//4
    , 'max_items': 15
    , 'max_iters': 200000
    , 'ratio_plate_size': 2
    , 'num_processors': 24
    , 'multiprocess': True
    , 'seed': 42
    , 'warm_start': False
    , 'heur_params': {'main_iter': 5, 'max_iter': 1000,
                      'temperature': 10000,
                      'max_candidates': 100, 'extra_jumbos': 0,
                      'cooling_rate': 0.0005,
                      'change_first': False, 'tolerance': None,
                      'try_rotation': 1,
                      'rotation_probs': [0.70, 0.30, 0, 0],
                      'rotation_tries': 2,
                      'level_probs': [0.3, 0.3, 0.3, 0.1],
                      'weights': {'space': 0.00001, 'seq': 40000, 'defects': 40000, 'wastes': 40000},
                      'cuts_prob': 0.5,
                      'clean_cuts_prob': 0.5
                      }
    , 'heur_remake': {
        'iterations_initial': 1000,
        'iterations_remake': 50,
        'rotation': [0.50, 0.50, 0, 0],
        'num_trees': [0.20, 0.20, 0.20, 0.20, 0.20],
        'options': ['best', 'partial', 'restart'],
        'probability': [0.05, 0.9, 0.05],
        'max_no_improve': 10,
        'prob_accept_worse': 0.1,
        'prob_accept_worse_def': 0.2,
        'prob_try_improve': 0.5,
        'prob_ref_is_last': 0.2
    }
    , 'heur_optim': {
        # 'try_rotation': True,
        # 'max_candidates': 5,
        # 'rotation_probs': [0.5, 0.5, 0, 0],
        # 'weights': {'space': 400, 'seq': 20000, 'defects': 40000, 'wastes': 40000}
    }
    , 'debug': False
    , 'graph': False
    # putting a tolerance on the heuristic seems horrible for finding good solutions.
    # putting change_first is also a very bad idea.
    # , 'cluster_tolerance': 50
}